src/formatDate.js

```diff
@@ -50,7 +50,53 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dates = date.split(fromFormat[fromFormat.length - 1]);
+  const filledInputFormat = {};
+  const outputFormat = [...toFormat];
+
+  outputFormat.length--;
+
+  for (let i = 0; i < dates.length; i++) {
+    filledInputFormat[fromFormat[i]] = dates[i];
+  }
+
+  for (let i = 0; i < outputFormat.length; i++) {
```

Comment:
it would be more clear to write `i < outputFormat.length - 1` rather than `outputFormat.length--;`